components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -642,11 +645,15 @@ private static Issuer getEntityIDAliasFromSP(String tenantDomain) throws Identit
                 IdentityApplicationConstants.Authenticator.SAML2SSO.NAME, IdentityApplicationConstants.Authenticator
                         .SAML2SSO.DESTINATION_URL_PREFIX));
         if (destinationURLs.size() == 0) {
-            String configDestination = IdentityUtil.getProperty(IdentityConstants.ServerConfig.SSO_IDP_URL);
-            if (StringUtils.isBlank(configDestination)) {
-                configDestination = IdentityUtil.getServerURL(SAMLSSOConstants.SAMLSSO_URL, true, true);
+
+            String destinationURL = resolveUrl(SAMLSSOConstants.SAMLSSO_URL, IdentityUtil.getProperty
```

Comment:
How about naming this as resolveIdpSsoUrl() or something?
resolveUrl() seems a bit generic and the method actually builds the IDP's SSO URL as the destination URL

Reply:
Well this method contract can accept any URL context and another value which it expects to be fed from the file configuration. Thus, shouldn't it be generic.